### PR TITLE
[stable/consul] Add cpu/memory limits to consul chart

### PR DESCRIPTION
**Pull request by bksteiny**
Hello there- 

I'd like to add cpu and memory limits to the Consul chart. 

Thanks
Chris

### CODE REVIEW

**Comment by bksteiny** [comment]
This wouldn't be a passive change. If you'd rather not bump the major version for this chart, I can make the changes in a way that are passive.

Please let me know your thoughts

**Comment by lachie83** [comment]
Thanks. Please bump major version

**Comment by jonstacks** [comment]
Seems like a rather small thing to bump the major version for. It looks like we could incorporate these changes in while doing so in a passive way. Would it be possible to wrap the limit assignment with something like:

```
{{- if .Values.resources.limits }}
limits:
  cpu: "{{ .Values.resources.limits.cpu }}"
  memory: "{{ .Values.resources.limits.memory }}"
{{- end }}
```

And for the other stuff we could maybe do something like: 

`{{ .Values.resources.requests.cpu | default .Cpu }}`

If not, I'd agree that we'd have to bump the major version.

File: stable/consul/templates/consul.yaml [hunk 1]
```diff
@@ -120,8 +120,11 @@ spec:
           containerPort: {{.Values.ConsulDnsPort}}
         resources:
           requests:
-            cpu: "{{ .Values.Cpu }}"
-            memory: "{{ .Values.Memory }}"
+            cpu: "{{ .Values.resources.requests.cpu }}"
+            memory: "{{ .Values.resources.requests.memory }}"
+          limits:
+            cpu: "{{ .Values.resources.limits.cpu }}"
+            memory: "{{ .Values.resources.limits.memory }}"
```
Comment (unguiculus) [inline stable/consul/templates/consul.yaml]: I'd suggest you change it like so:

```yaml
        resources:
{{ toYaml .Values.resources | indent 10 }}
```

File: stable/consul/values.yaml [hunk 2]
```diff
@@ -18,8 +18,13 @@ Replicas: 3
 Image: "consul"
 ImageTag: "0.8.3"
 ImagePullPolicy: "Always"
-Cpu: "100m"
-Memory: "256Mi"
+resources:
```
Comment (unguiculus) [inline stable/consul/values.yaml]: With above change do it like so:

```yaml
resources: {}
#  requests:     
#    cpu: "100m"
#    memory: "256Mi"
#  limits:
#    cpu: "500m"
#    memory: "512Mi"
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
We've started recommending not to specify defaults for resources. This should be a conscious choice of the user and makes it easier to test charts on smaller systems, such as Minikube.

File: stable/consul/templates/consul.yaml (hunk 1 above)
Comment (bksteiny) [reply stable/consul/templates/consul.yaml]: Nice suggestion

File: stable/consul/values.yaml (hunk 2 above)
Comment (bksteiny) [reply stable/consul/values.yaml]: 👌

File: stable/consul/README.md [hunk 3]
```diff
@@ -32,14 +32,16 @@ The following tables lists the configurable parameters of the consul chart and t
...
 | `EncryptGossip`         | Whether or not gossip is encrypted    | `true`                                                     |
-| `Memory`                | container requested memory            | `512Mi`                                                    |
 | `Storage`               | Persistent volume size                | `1Gi`                                                      |
 | `StorageClass`          | Persistent volume storage class       | `nil`                                                      |
 | `HttpPort`              | Consul http listening port            | `8500`                                                     |
+| `resources.limits.cpu`  | container limited cpu                 | `500m`                                                     |
+| `resources.limits.memory` | container limited memory            | `512Mi`                                                    |
+| `resources.requests.cpu`| container requested cpu               | `100m`                                                     |
+| `resources.requests.memory` | container requested memory        | `256Mi`                                                    |
```
Comment (unguiculus) [inline stable/consul/README.md]: Should be just one resources row (with proper spacing, of course).

```
| `Resources` | Pod resource requests and limits | `{}` |
```

File: stable/consul/templates/consul.yaml [hunk 4]
```diff
@@ -119,9 +119,7 @@ spec:
           protocol: "UDP"
           containerPort: {{.Values.ConsulDnsPort}}
         resources:
-          requests:
-            cpu: "{{ .Values.Cpu }}"
-            memory: "{{ .Values.Memory }}"
+{{ toYaml .Values.resources | indent 10 }}
```
Comment (unguiculus) [inline stable/consul/templates/consul.yaml]: Make it start with an upper case letter for consistency (even though that wouldn't match our best practices).

`{{ toYaml .Values.Resources | indent 10 }}`

File: stable/consul/values.yaml [hunk 5]
```diff
@@ -18,8 +18,13 @@ Replicas: 3
 Image: "consul"
 ImageTag: "0.8.3"
 ImagePullPolicy: "Always"
-Cpu: "100m"
-Memory: "256Mi"
+resources: {}
```
Comment (unguiculus) [inline stable/consul/values.yaml]: `Resources: {}`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by bksteiny** [comment]
Hi @unguiculus - I've added your suggestions

**Comment by unguiculus** [comment]
Tests pass on my GKE cluster.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**